src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.c

```diff
@@ -775,8 +787,10 @@ static void create_rr_locked(grpc_exec_ctx *exec_ctx, glb_lb_policy *glb_policy,
 /* glb_policy->rr_policy may be NULL (initial handover) */
 static void rr_handover_locked(grpc_exec_ctx *exec_ctx,
                                glb_lb_policy *glb_policy) {
-  GPR_ASSERT(glb_policy->serverlist != NULL &&
-             glb_policy->serverlist->num_servers > 0);
+  GPR_ASSERT((glb_policy->fallback && glb_policy->fallback_addresses != NULL &&
```

Comment:
Must these two asserts both hold?

- We can just have valid fallback, if we don't have a valid server list because we can't reach any balancer.
- We can just have valid server list, because fallback is unnecessary then.